packages/http/src/client.ts

```diff
@@ -0,0 +1,141 @@
+import { IPrismOutput } from '@stoplight/prism-core';
+import { IHttpOperation } from '@stoplight/types';
+// @ts-ignore
+import logger from 'abstract-logging';
+import { getOrElse } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { getStatusText } from 'http-status-codes';
+import { defaults } from 'lodash';
+import { parse as parseQueryString } from 'querystring';
+import { parse as parseUrl } from 'url';
+import { createInstance } from '.';
+import { forwarder } from './forwarder';
+import getHttpOperations, { getHttpOperationsFromResource } from './getHttpOperations';
+import { mocker } from './mocker';
+import { router } from './router';
+import { IHttpConfig, IHttpRequest, IHttpResponse, IHttpUrl } from './types';
+import { validator } from './validator';
+
+async function createClientFromResource(specFile: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
+  const resources = await getHttpOperationsFromResource(specFile);
+  return createClientFromOperations(resources, defaultConfig);
+}
+
+async function createClientFromString(spec: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
+  const resources = await getHttpOperations(spec);
+  return createClientFromOperations(resources, defaultConfig);
+}
+
+function createClientFromOperations(resources: IHttpOperation[], defaultConfig: IHttpConfig) {
+  const obj = createInstance(defaultConfig, {
+    logger,
+    router,
+    forwarder,
+    validator,
+    mocker,
+  });
+
+  const request: RequestFunction = async (url, input, config) => {
```

Comment:
could `request` be moved out of `createClientFromOperations`? Or, is there any reason to keep it here?

Reply:
The idea is to make sure nobody will use this function, if not `createClientFromOperations`

The rationale is that no matter how you want to effectively create a client; all the ways MUST eventually call this function. By making the `request` function private here, nobody can create a client in another way. Does that make sense?

Reply:
@XVincentX, hm.. why making sure of that? Isn't only exporting what you are exporting enough? Moving out the inner functions and keeping them private (by not exporting them), gives us slimmer functions and I believe better performance (no hard facts on this though).